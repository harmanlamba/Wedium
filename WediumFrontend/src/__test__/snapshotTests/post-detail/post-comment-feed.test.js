import React from 'react';
import renderer from 'react-test-renderer';
import {
  shallow
} from 'enzyme';

import PostCommentFeed from '../../../components/post-detail/comment/post-comment-feed';

it('snapshot postCommentFeed default load', () => {
  const user = {
    firstName: 'null',
    isAuthenticated: false,
    jwtToken: 'null',
    lastName: 'null',
    pictureUri: 'null',
    username: 'null',
  };

  const comments = [];

  const setRouteLeaveHook = jest.fn();

  let wrapper = shallow( <
    PostCommentFeed.WrappedComponent user = {
      user
    }
    comments = {
      comments
    }
    params = {
      {
        router: setRouteLeaveHook,
      }
    }
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('snapshot postCommentFeed with comments load', () => {
  const user = {
    firstName: 'null',
    isAuthenticated: false,
    jwtToken: 'null',
    lastName: 'null',
    pictureUri: 'null',
    username: 'null',
  };

  const comments = [{
    body: 'This is a comment',
    commentId: 1,
    commentTypeId: 1,
    date: '2020-05-21T21:56:36.933',
    inverseParentComment: [],
    parentCommentId: null,
    postId: 123,
    userId: 123,
    userName: 'Bob Shaw',
  }, ];

  const setRouteLeaveHook = jest.fn();

  let wrapper = shallow( <
    PostCommentFeed.WrappedComponent user = {
      user
    }
    comments = {
      comments
    }
    params = {
      {
        router: setRouteLeaveHook,
      }
    }
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('snapshot postCommentFeed with user authenticated load', () => {
  const user = {
    firstName: 'Bob',
    isAuthenticated: true,
    jwtToken: 'jwtToken',
    lastName: 'Jenkins',
    pictureUri: 'somePhoto',
    username: 'Bob Jenkins',
  };

  const comments = [{
    body: 'This is a comment',
    commentId: 1,
    commentTypeId: 1,
    date: '2020-05-21T21:56:36.933',
    inverseParentComment: [],
    parentCommentId: null,
    postId: 123,
    userId: 123,
    userName: 'Bob Shaw',
  }, ];

  const setRouteLeaveHook = jest.fn();

  let wrapper = shallow( <
    PostCommentFeed.WrappedComponent user = {
      user
    }
    comments = {
      comments
    }
    params = {
      {
        router: setRouteLeaveHook,
      }
    }
    />
  );

  expect(wrapper).toMatchSnapshot();
});