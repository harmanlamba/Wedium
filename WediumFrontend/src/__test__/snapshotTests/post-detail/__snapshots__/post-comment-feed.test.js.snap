// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot postCommentFeed default load 1`] = `
<PostCommentFeed
  classes={
    Object {
      "commentCard": "PostCommentFeed-commentCard-5",
      "expand": "PostCommentFeed-expand-3",
      "expandOpen": "PostCommentFeed-expandOpen-4",
      "grid": "PostCommentFeed-grid-2",
      "root": "PostCommentFeed-root-1",
    }
  }
  comments={Array []}
  params={
    Object {
      "router": [MockFunction],
    }
  }
  user={
    Object {
      "firstName": "null",
      "isAuthenticated": false,
      "jwtToken": "null",
      "lastName": "null",
      "pictureUri": "null",
      "username": "null",
    }
  }
/>
`;

exports[`snapshot postCommentFeed with comments load 1`] = `
<PostCommentFeed
  classes={
    Object {
      "commentCard": "PostCommentFeed-commentCard-5",
      "expand": "PostCommentFeed-expand-3",
      "expandOpen": "PostCommentFeed-expandOpen-4",
      "grid": "PostCommentFeed-grid-2",
      "root": "PostCommentFeed-root-1",
    }
  }
  comments={
    Array [
      Object {
        "body": "This is a comment",
        "commentId": 1,
        "commentTypeId": 1,
        "date": "2020-05-21T21:56:36.933",
        "inverseParentComment": Array [],
        "parentCommentId": null,
        "postId": 123,
        "userId": 123,
        "userName": "Bob Shaw",
      },
    ]
  }
  params={
    Object {
      "router": [MockFunction],
    }
  }
  user={
    Object {
      "firstName": "null",
      "isAuthenticated": false,
      "jwtToken": "null",
      "lastName": "null",
      "pictureUri": "null",
      "username": "null",
    }
  }
/>
`;

exports[`snapshot postCommentFeed with user authenticated load 1`] = `
<PostCommentFeed
  classes={
    Object {
      "commentCard": "PostCommentFeed-commentCard-5",
      "expand": "PostCommentFeed-expand-3",
      "expandOpen": "PostCommentFeed-expandOpen-4",
      "grid": "PostCommentFeed-grid-2",
      "root": "PostCommentFeed-root-1",
    }
  }
  comments={
    Array [
      Object {
        "body": "This is a comment",
        "commentId": 1,
        "commentTypeId": 1,
        "date": "2020-05-21T21:56:36.933",
        "inverseParentComment": Array [],
        "parentCommentId": null,
        "postId": 123,
        "userId": 123,
        "userName": "Bob Shaw",
      },
    ]
  }
  params={
    Object {
      "router": [MockFunction],
    }
  }
  user={
    Object {
      "firstName": "Bob",
      "isAuthenticated": true,
      "jwtToken": "jwtToken",
      "lastName": "Jenkins",
      "pictureUri": "somePhoto",
      "username": "Bob Jenkins",
    }
  }
/>
`;
